login-utils/login.c

```diff
@@ -260,18 +256,19 @@ static void process_title_init (int argc, char **argv)
 		argv0 = argv;
 }
 
-static void process_title_update (const char *prog, const char *txt)
+static void process_title_update(const char *username)
 {
         size_t i;
-        char buf[SPT_BUFSIZE];
+        const char prefix[] = "login -- ";
+        char buf[sizeof(prefix) + LOGIN_NAME_MAX];
 
         if (!argv0)
                 return;
 
-	if (strlen(prog) + strlen(txt) + 5 > SPT_BUFSIZE)
+	if (sizeof(buf) < (sizeof(prefix) + strlen(username) + 1))
 		return;
 
-	sprintf(buf, "%s -- %s", prog, txt);
+	sprintf(buf, "%s%s", prefix, username);
```

Comment:
Please, use snprintf() here -- I know there is the if() where it checks for the size, but sprintf() works like a red flag for static analyzers ... :-)

Reply:
Changed to `sprintf()` in d51538190134e490fa1a26b7ea6ac4956bf64076